v2/git-initializer/e2e/e2e.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"time"
+
+	"github.com/ghodss/yaml"
+
+	"github.com/brigadecore/brigade/sdk/v2"
+	"github.com/brigadecore/brigade/sdk/v2/authx"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/sdk/v2/restmachinery"
+	"github.com/pkg/errors"
+)
+
+type testcase struct {
+	name          string
+	projectFile   string
+	terminalPhase core.WorkerPhase
+}
+
+var testcases = []testcase{
+	{
+		name:          "GitHub - Public Repo - https",
+		projectFile:   "github-public-https.yaml",
+		terminalPhase: core.WorkerPhaseSucceeded,
+	},
+	// TODO: once we have mech for injecting private ssh key in secure manner
+	// {
+	// 	name:          "GitHub - Private Repo - ssh",
+	// 	projectFile:   "github-private-ssh.yaml",
+	// 	terminalPhase: core.WorkerPhaseSucceeded,
+	// },
+	{
+		name:          "GitHub - Public Repo - submodules",
+		projectFile:   "github-init-submodules.yaml",
+		terminalPhase: core.WorkerPhaseSucceeded,
+	},
+	{
+		name:          "GitHub - Malformed URL",
+		projectFile:   "github-malformed-url.yaml",
+		terminalPhase: core.WorkerPhaseFailed,
+	},
+}
+
+func main() {
+	for _, tc := range testcases {
+		err := runTest(tc)
+		if err != nil {
+			log.Fatalf("Test %q failed: %s", tc.name, err)
+		}
+	}
+}
+
+func runTest(tc testcase) error {
+	ctx, cancel :=
+		context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+
+	// TODO: send in/parameterize this value
+	apiServerAddress := "https://localhost:7000"
+	// TODO: send in/parameterize this value
+	rootPassword := "F00Bar!!!"
+	apiClientOpts := &restmachinery.APIClientOptions{
+		AllowInsecureConnections: true,
+	}
+	filename := fmt.Sprintf("../testdata/%s", tc.projectFile)
+
+	log.Printf("Running test %q...\n", tc.name)
+
+	projectBytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return errors.Wrapf(err, "error reading project file %s", filename)
+	}
+
+	if projectBytes, err = yaml.YAMLToJSON(projectBytes); err != nil {
+		return errors.Wrapf(err, "error converting file %s to JSON", filename)
+	}
+
+	project := core.Project{}
+	if err = json.Unmarshal(projectBytes, &project); err != nil {
+		return errors.Wrapf(err, "error unmarshaling project file %s", filename)
+	}
```

Comment:
I have two mutually exclusive thoughts for how to improve these tests a smidge.

1. These tests are falling just short of being truly e2e because the CLI isn't being used to drive the tests. Executing `brig project create -f <path to test case>` would spare you from reading the file and converting it to JSON because the CLI would handle all of that for you. This would be more truly e2e, but it would also make these tests way uglier, so I don't entirely mind if we _don't_ do this right now, which takes me to option 2...

2. Since you're already using the SDK, consider not bothering with files for the test cases. You can enumerate the test cases _directly_ as `core.Project` objects at the top of this file (or in a separate file). It would spare you from messing with reading files and converting the yaml to json.

Reply:
Yea, I like 2.

Considering 1. and the comment below (to use the go testing framework), perhaps we move this to `main_test.go` and relegate proper e2e (using the cli as mentioned, etc.) to an effort down the road.  WDYT?

Reply:
Yes. I like. Call this "integration" instead of e2e and we'll kick the e2e can down the road.

Reply:
Ok, pushed a commit addressing the other comments; will proceed to working on this change next.